### ISSUE TITLE
Collaboration on checklist project

### ISSUE THREAD

**Issue by patrickwalkowicz** [codingforeveryone/js#27]
Thanks for today's hangout! For those who couldn't make it, we've decided to collaborate on manageable projects to practice with. The first one we chose is building a front of a checklist application. So this would require HTML + CSS + JavaScript/jQuery.

What I think the application should be able to do:
- Add checklist items typed by user to the "To Do" list
- Move checked items to "Completed" list (or delete them altogether after checking them off "To Do" list)

@MyPitit, @FrancescoPalma, @daymos, @thevalent, @Bektaz, @JMurphyWeb, what do you think?

**Comment by MyPitit** [comment]
Hi all! It was very nice meeting all of you today. 
@patrickwalkowicz that sounds very good to me, I will do some "research" this week to see how/what we can add. We can discuss all our ideas next Sunday before getting our hand dirty with code. 

Regarding the backend part, I suggest this service. https://www.firebase.com/pricing.html We have up to 1GB free and looks very suitable for us.  

Let me know what do you think team.

**Comment by patrickwalkowicz** [comment]
@MyPitit [Firebase](https://www.firebase.com/) looks good, I'll go throught the [examples](https://www.firebase.com/tutorial/#session/f7c65x84zt6) this week for better understanding.

I suggest implementing the front part first and then adding code for pushing entries to the database.

**Comment by thevalent** [comment]
I agree with all the above. I can't wait to start getting this roll. Firebase sounds very good to me, never heard about it before, you are spot on on that Mireia!

**Comment by Bektaz** [comment]
hi mates I also agree what about removing items from 'to do' list once the item is chosen.
I have also heard Firebase for the 1st time, what do we need to know to start learning it? are there any prerequisites except js?

**Comment by MyPitit** [comment]
@Bektaz no prerequisites, just  need to register and start using their library, easy :P Don't worry about that for now. We have to figure it out first how to build a front of a checklist application :)

**Comment by Bektaz** [comment]
thanks @patrickwalkowicz I will try it :+1:

**Comment by JMurphyWeb** [comment]
Love this idea. How do we know what tasks we should each do? Do I understand correct that we want front end done this week and we can look into back end next week?

Today I have been trying to make a basic chrome app, this project could eventually be developed to work as a chrome app and released on the app store.

**Comment by MyPitit** [comment]
Hi all! I have a suggestion, we can use the Kanban system to track all the tasks. It's a very good way to organise all the tasks, trak the progress, team work, etc. I use Trello for that, https://trello.com it's very simple to use. I already create a new board for this project. If this sounds good to you guys I can sent the invitation to all of you :)
@JMurphyWeb we don't really have assigned any specific task at the moment, we will discuss this next Saturday I guess. I have done some research to see how we can create this project. 
I would love to hear more about your new project, sounds very cool :+1:

**Comment by FrancescoPalma** [comment]
Hi everyone! Sorry for my delayed reply, I've been very busy.
@patrickwalkowicz Thank you again for opening this issue and I agree 100% with you about the structure of the application you propose. 
@MyPitit Thank you for the research work you're doing. I've never heard of Firebase but very willing to learn something new :+1: I've got Trello already, it's a great application for project mangament, but never used it yet. I agree on using it, send me the invitation whenever you want :)

**Comment by MyPitit** [comment]
@FrancescoPalma I already send the invitation to you and also to @Bektaz, @daymos, @thevalent, @patrickwalkowicz. If I miss someone here please give me your email and I will send the invitation a soon as possible :)

**Comment by MyPitit** [comment]
@JMurphyWeb I also send the invitation to you

**Comment by patrickwalkowicz** [comment]
@MyPitit thanks for Trello invitation!

I've been thinking about how could we collaborate on a practical level. I was looking for a solution that would enable us to leverage the team aspect and have everybody actively participating in actual coding. Because of the project being small, I thought it's hard to actually delegate tasks and work on them at the same time. We'd be stepping on each other's toes trying to edit the same code and it would be hard to coordinate. However, that approach would work well in more elaborate project where each team member could work on one separate function/module.

On the other end of spectrum is us working together at the same time. However, this means that one person would be actively coding at a time, while others would give suggestions etc. The disadvantage is that we'd have one person that is actually active and the rest would be passive to some extent.

So I came up with an alternative that would enable us to leverage team work setting and at the same time let each of us work actively on code. My idea is that we could do a group discussion/briefing first and then divide into pairs. From what I've seen the concept of pair programming is quite popular in the industry. So each pair would create their version of the project and that way each of us would participate actively at all times. Each pair could work through [Team Viewer](https://www.teamviewer.com/en/index.aspx) which is a remote desktop experience with voice / video chat on top (so it's as close to being in the same room as possible). The software is free for personal use.

What do you think?

**Comment by MyPitit** [comment]
@patrickwalkowicz I love your idea, I agree 100% with you, this sounds very doable if we want to code.
Have you looked at Etherpad? I never used it before but my partner really recommended it. http://etherpad.org/

**Comment by patrickwalkowicz** [comment]
@MyPitit this one looks good! Do you know if you can get it to colour the syntax or is it just a plain text editor? It looks like a good alternative for Team Viewer, however with TV you get to use your favourite tools as it's based on remote desktop experience.

Another thought about splitting into pairs. It would help us work around our schedules because only 2 people would have to arrange a meeting at one time, instead of all 6 of us.

**Comment by daymos** [comment]
I like the idea of pair programming. I think I read something about it before. The second programmer  has to understand what the first programmer has done  before he can add his stuff. So everybody is kept up-to-date.

**Comment by FrancescoPalma** [comment]
Both TeamViewer and EtherPad look effective, we might try one for the next session and then the other one so we understand which one we are more comfortable with. I like the idea of pair programming, it might be a bit difficult at the beginning, especially for beginners (like me) but it's going to be really rewarding in the long-term since I think it is an important learning tool. On this matter, we can find more information about this feature [here](https://github.com/foundersandcoders/playbook/blob/master/pair-programming.md) and [here](http://www.wikihow.com/Pair-Program).

**Comment by thevalent** [comment]
That's brilliant either of the options you mentioned guys is alright for me. I already have team viewer installed and having two screens on my pc I could see what the person I will be paired with will be doing. I believe that everyone has its own pace and since I never programmed paired with anyone before I think I'll need some time before being effective but it sounds really appealing to me!

**Comment by JMurphyWeb** [comment]
@patrickwalkowicz pair programming is a great idea. Due to time differences it would make sense for me and @Bektaz to join forces. I will look into trello now. What is the time frame are we working with for this project?

**Comment by Bektaz** [comment]
hi guys, brilliant ideas especially pair programming i am for it with my both hands up :smile: @patrickwalkowicz many thanks for offering that idea. Fully agree with @JMurphyWeb about time differences and pairing with him. I dont have any experience in pair programming either thus not used EtherPad and TV in this matter, but used TV just for OP system administrative issues I liked it, hope it works fine for programming as well. So are we going to meet on Saturday to finalize and start working on Sunday, how it is going to be?

**Comment by FrancescoPalma** [comment]
I suggest a suitable plan might be as follows;
1) we decide the pairs here,
2) we refer to the GoogleDoc created by @MyPitit for documentation, research and data. Everyone is welcome to add stuff,
3) the pairs make their own arrangements using platforms and tools of their choice,
4) between Saturday and Sunday (according to the pairs' arrangements and plans) the actual building happens,
5) we set a time for the general meeting (let's say UK afternoon?) on G-Hangouts,
6) each pair shows what they've done, how, what tools, documentation etc. Anything related to the project,
7) we exchange ideas and feedback on each others' checklist based on criteria such as readability, efficiency, 
8) finally, we agree on the next project for the following week setting the first arrangements and tasks (i.e. open a new issue here, start doing research, add suggestions and thoughts etc.).

What do you guys think about this? I might've missed something, if you guys have different plans or ideas in mind, speak up :smiley:

**Comment by patrickwalkowicz** [comment]
That's a good one, @FrancescoPalma! How about having a meeting before we actually start coding to discuss implementation ideas/tools to use (for example at 10AM on Sunday). If we keep it short (say 40 minutes) then we'll have plenty of time to do actual coding later on. Then we could meet a week later to briefly present the code *and* choose the next project. 

I'm proposing this alternative because as mentioned during the last hangout, some of the guys preferred meeting on Sunday due to lack of free time on Saturday. By doing it that way those of us who can't work on Saturday won't be pressed on time to present their code on Sunday. What do you think?

**Comment by FrancescoPalma** [comment]
@patrickwalkowicz Agreed :+1: actually it is a better proposal from a time perspective. No rush, better allow more time by using also the weekdays (or any other time in between) to polish the code.

**Comment by daymos** [comment]
@patrickwalkowicz @FrancescoPalma I agree in allowing some more time. I think each of us has a different skill level at this point. To work well together, especially in a 2 men team, I think there should be a common knowledge at least on the subject we are going to work on. To build this "common ground" I suggest everybody  who fells less experience in web development with a back-end ( I include myself ) to have a good grasp on the topics covered by this code accademy course: https://www.codecademy.com/en/skills/make-an-interactive-website.
I think it could be a good starting point ( and is part of FEC requirements )

**Comment by JMurphyWeb** [comment]
@daymos i assumed that Codecademy track was primarily front end? Ive been meaning to do it anyway so maybe now is a good time.  

I am away and unavailable between Saturday and Tuesday so if it works out with @Bektaz then maybe him and i could start something today or tomorrow?

**Comment by Bektaz** [comment]
hi @daymos thanks for the link you shared, apart of that do remember during our last hangout you told there is another js course in codeacademy which is not shown usually on the page, could you share with the link? @JMurphyWeb I agree with you we could start discussing even today if you are not available from Saturday, just let me know specific time

**Comment by daymos** [comment]
@JMurphyWeb You are right that is 100% front-end. If someone has never done a webpage with a db, I doubt they can start  immediately with that. But that would give sufficient frontend skills to work on the project anyways.  Somebody on the chat spoke about udemy, so I found this: https://www.udemy.com/nodejs-in-30/ 
It should explain how to do all the steps of backend including deployment. And it uses express framework which is a good one in my opinion. 
With these 2 resources any pair should be able to complete the project.

**Comment by MyPitit** [comment]
All the proposals heres are looking very good. How are we going to do the teams of two? Shall we discuss this on Sunday or before Sunday?

**Comment by daymos** [comment]
@Bektaz this is the one about recursion

**Comment by daymos** [comment]
@MyPitit I have to work this Sunday. I am available any other day.

**Comment by FrancescoPalma** [comment]
I think we can decide the pairs here even now (or anyway before Sunday). So, if @JMurphyWeb  and @Bektaz are going to pair. We are 5 left (me, @patrickwalkowicz, @thevalent , @MyPitit , @daymos ) so we are not even. If we find someone else willing to join it's great, otherwise we can just make 2 pairs and the one left can try to do on its own (if he/she has a more advanced knowledge) or can try to collaborate with one of the pairs as a third member. To me, we can make one pair of three as well, no issue. Let me know your thoughts guys about this :)

**Comment by Bektaz** [comment]
hi guys, @daymos thanks very much for the link, about lacking member to build complete pairs with two i would suggest @toast565 if he would like to join. he is the new member in this repo. If he has an interest then we would become even.

**Comment by MyPitit** [comment]
@daymos I'm wfh, so I'm available at any time :) @FrancescoPalma, like you said if we can do the teams before Sunday I think it will be better.

**Comment by patrickwalkowicz** [comment]
@Bektaz I agree, would be cool if @toast565 could join us as well and we'd have an even number of participants. Are you up to that, @toast565?

**Comment by toast565** [comment]
Hi @patrickwalkowicz, thanks for asking me. I'm in principle very up for this, though I have somewhat limited time. Count me in for now - I will read up on the project tonight and see if it looks too much for me to commit to.

**Comment by patrickwalkowicz** [comment]
Very cool, @toast565 :+1:  
Please keep in mind that it's learn-as-you-go, the idea is to learn how to figure things out and learn in real time. Also, the project is pretty straightforward so we can get it done in a short period of time. 
We'll discuss the whole thing before we start so we're all on the same page. 
Let us know :smile:

**Comment by toast565** [comment]
Had a read, and yeah this looks great, I'm in! Like others I'm busy Saturday but free all of Sunday. I'm completely unfamilliar with back-end stuff but hopefully there would be a minimal amount of that, and we can pool our knowledge! 

I currently don't have access to the JS Hangout gitter yet, so if someone could add me that would be great. Also I think I saw something about a GoogleDoc? Happy to be paired with anyone, though I think in this crowd I'm pretty inexperienced.

**Comment by FrancescoPalma** [comment]
@toast565 then, warm welcome to our group! :tada: I think @MyPitit can send you an invitation on Trello and on the GoogleDoc, if you provide us with your mail we can add you in our personal contacts on GoogleHangouts so we'll be able to let you in the next multichat. Also, on Gitter you're welcome to join the chatroom [here](https://gitter.im/codingforeveryone/JSHangout) ;)

**Comment by FrancescoPalma** [comment]
@toast565 and don't worry about inexperience, there are more beginners than you think around you, we all have to learn and implement our knowledge. The best thing to do is to try and write as much code as we can to polish our skills step by step.

**Comment by toast565** [comment]
Cheers for the welcome! I think I might need some kind of invitation to the chatroom as well though, that link gives me a 404 'This is not the chat you're looking for...'

**Comment by patrickwalkowicz** [comment]
@toast565 great!

I've just tried adding you to the Gitter chatroom but it looks like you're not in codingforeveryone group yet. In order to join the hangout chat group, you need to be added to codingforeveryone first. Please make sure to submit the [first part of the application](http://www.foundersandcoders.com/academy/apply1.html) so you're added to codingforeveryone. 
You just need to provide your GitHub and Twitter username so you can be added by F&C guys.

**Comment by JMurphyWeb** [comment]
@bektaz i cannot find your email. Could you let me know if you are free today from 6am to 11am gmt? 
Jmurphy.web@gmail.com

I was having a look around and it seems a Combination of Heroku and node is a common backend hosting solution with lots of supporting documents online. Is there any reason why this wouldnt work for us?

**Comment by MyPitit** [comment]
@toast565 Welcome! 
I need your email for the google doc invitation please.

**Comment by toast565** [comment]
@MyPitit oturnermajor@gmail.com - thanks!

**Comment by MyPitit** [comment]
Thanks @toast565, you will receive the invitation shortly.

**Comment by Bektaz** [comment]
hi @JMurphyWeb sorry for delay I was busy a bit. Probably you will not participate for tomorrow's hangout, so if you can find some short time for discussion email me at berekettaz@gmail.com to set time for discusson of details of the project (notification of email comes straight away, so that would be good option to reply you fast). May be while you are busy up to Tuesday I might be doing something with our project to be in time. Guys what time are we going to do hangout tomorrow?

**Comment by FrancescoPalma** [comment]
Hi all! We can meet on Hangouts at 10am UK time. Is that ok?

**Comment by patrickwalkowicz** [comment]
@Bektaz @FrancescoPalma Cool! I believe everybody was ok with 10AM BST.

**Comment by daymos** [comment]
Hi all, unfortunately I have to work the whole day tomorrow. If there is someone who wants to work on the toDo list project during the week instead of tomorrow, I could be paired with this person.

**Comment by FrancescoPalma** [comment]
@daymos @MyPitit @toast565 two pairs had been set. We are the 4 left, we can make two pairs. One with @daymos who can work during the week, the other two starting from tomorrow. I can tomorrow, whereas I am not sure if I can do that during the week other than updates and quick tasks at evening.

**Comment by MyPitit** [comment]
Hi all, I can work during the week and also on the weekend. I don't have any preferences. :)

**Comment by MyPitit** [comment]
So I can pair with @daymos if anyone here can't do the project during the week.

**Comment by daymos** [comment]
@MyPitit ok so we are matched!

**Comment by Bektaz** [comment]
10 AM is fine for me :+1:

**Comment by toast565** [comment]
Which leaves @FrancescoPalma and myself :) Talk to you all 10am tomorrow!

**Comment by FrancescoPalma** [comment]
@toast565 Great! See you all tomorrow guys!

**Comment by Katbow** [comment]
Hey, is this still being worked on? If so, which repository is being used for it? Thanks!

**Comment by toast565** [comment]
Hey @Katbow, this ended up being pairs of us working on projects in our own repos. Pretty sure everyone's done with it now though - can probably close the issue.

**Comment by Katbow** [comment]
Okay, thanks for the update. Closing issue.
